containers/src/Kokkos_DynamicView.hpp

```diff
@@ -195,11 +195,13 @@ struct ChunkedArrayManager {
   void deep_copy_to(
       const ExecutionSpace& exec_space,
       ChunkedArrayManager<OtherMemorySpace, ValueType> const& other) const {
-    if (other.m_chunks != m_chunks) {
-      Kokkos::Impl::DeepCopy<OtherMemorySpace, MemorySpace, ExecutionSpace>(
-          exec_space, other.m_chunks, m_chunks,
-          sizeof(pointer_type) * (m_chunk_max + 2));
-    }
+    // use of ad-hoc unmanaged views
+    Kokkos::deep_copy(
+        exec_space,
+        Kokkos::View<value_type*, OtherMemorySpace>(
+            reinterpret_cast<value_type*>(other.m_chunks), m_chunk_max + 2),
+        Kokkos::View<value_type*, MemorySpace>(
+            reinterpret_cast<value_type*>(m_chunks), m_chunk_max + 2));
```

Comment:
Why is the `interpret_cast` the right thing to do here?

Reply:
`m_chunks` is a double pointer that is treated as `void *` by `Dynk::DeepCopy`. `Kokkos::View`, even unmanaged, requires to set the appropriate rank and dimensions, the later being unknown. So I use `reinterpret_cast` to force treating `m_chunks` as 1-rank data. As the view is temporary and only used for the call of `deep_copy`, I think it is OK to do so.

Reply:
If I'm not mistaken, the original call did something like
```C++
memcpy(reinterpret_cast<pointer_type*>(&other.m_chunks), reinterpret_cast<pointer_type*>(&m_chunks), sizeof(pointer_type) * (m_chunk_max + 2));
```
(where the reinterpret_cast is irrelevant of course)
while the new implementation does
```C++
memcpy(reinterpret_cast<value_type*>(&other.m_chunks), reinterpret_cast<value_type*>(&m_chunks), sizeof(value_type) * (m_chunk_max + 2));
```
and these would only do the same if `sizeof(value_type) == `sizeof(pointer_type)`. In other words, the Views data type should be `pointer_type` (which is problematic since int** would be treated as a two-dimensional View) or something with the same size like `uintptr_t`.

Reply:
To clarify things out, `pointer_type` is the same as `value_type*`, and `m_chunks` is a `pointer_type*`.

I see what you mean, so my PR is very likely copying a wrong amount of data (since `sizeof(value_type*) == sizeof(pointer_type)`), then?

Reply:
> I see what you mean, so my PR is very likely copying a wrong amount of data (since `sizeof(value_type*) == sizeof(pointer_type)`), then?

That's what it looks like to me.

Reply:
Do you have any idea on how to fix that?

Reply:
Use `uintptr_t` for the `View`'s data type (and reinterpret_cast to `uintptr_t*` instead of `value_type*`).

Reply:
There is no guarantee that `value_type` is an `int`, since it's template parameter. In the tests I see, it's a `double`.

Reply:
`uintptr_t` does have the same size as a pointer. It doesn't matter what the type pointed to is. Also, round-trip conversion between a pointer and `uintptr_t` is fine and that's what we are effectively doing here. The `memcpy` never tries to interpret values stored in `uintptr_t`.

Reply:
Ok, I didn't know that. Seems like it works.

Reply:
Ew, the code segfaults now...

Reply:
Fixed.